x-pack/plugins/alerts/common/alert.ts

```diff
@@ -15,6 +15,28 @@ export interface IntervalSchedule extends SavedObjectAttributes {
   interval: string;
 }
 
+// for the `typeof ThingValues[number]` types below, become string types that
+// only accept the values in the associated string arrays
+export const AlertExecutionStatusValues = ['ok', 'active', 'error', 'unknown'] as const;
+export type AlertExecutionStatuses = typeof AlertExecutionStatusValues[number];
+
+export const AlertExecutionStatusErrorReasonValues = [
+  'read',
+  'decrypt',
+  'execute',
+  'unknown',
+] as const;
+export type AlertExecutionStatusErrorReasons = typeof AlertExecutionStatusErrorReasonValues[number];
+
+export interface AlertExecutionStatus {
+  status: AlertExecutionStatuses;
+  date: Date;
```

Comment:
it does feel kind of "context free", viewed here, but it is a property of the `executionStatus` object, so I thought using `date` by itself would be fine; ie, you're always be referencing this piece of data as `executionStatus.date`.  Adding additional context on `date` itself seemed like overkill and overly wordy. 🤔

Reply:
well, it isn't obvious to me what this date actually is 🤷 
I won't block on this, but my feeling is that we're adding to the cognitive load by not being clear what his date actually means.
I'm guessing it means "lastUpdate" of the status, but I'm still not 100% sure and to me that's a reason to clarify.

Reply:
But you know what the `status` is?  Should we change it to `lastExecutionStatus` as well? :-) I think that was my thinking in trying to not add more context here to the prop names, when it feels like it's implied by it's containing property &| type.

Contextually, how people would end up accessing this, would look like the following for the two variants:

- `alert.executionStatus.date`
- `alert.executionStatus.lastExecutionDate`

I don't have really strong feels, just trying to cut down verbosity / ceremony / overkill where not actually needed.

But I just thought of a decent reason to do this - if for some reason we add some other date to this structure later, THEN it will certainly be confusing what the un-prefixed date would be.

So, I think `lastExecutionDate` prolly works best for me, since that exactly describes it.

Reply:
Current branch is using `lastExecutionDate` - thanks for prodding on this Gidi!